File: kotlin/src/main/kotlin/codingdojo/Equipment.kt
package codingdojo

class Equipment(
    // TODO add a ring item that may be equipped
    //  that may also add damage modifier
    val leftHand: Item,
    val rightHand: Item,
    val head: Item,
    val feet: Item,
    val chest: Item,
    val wing: Item,
    val tail: Item
) {
    val baseDamage: Int
    get() {
        return leftHand.baseDamage +
            rightHand.baseDamage +
            head.baseDamage +
            feet.baseDamage +
            chest.baseDamage +
                wing.baseDamage + tail.baseDamage
    }
}

File: swift/FantasyBattle/Equipment.swift
protocol EquipmentProtocol {
    var leftHand: Item { get }
    var rightHand: Item { get }
    var head: Item { get }
    var feet: Item { get }
    var chest: Item { get }
    func getBaseDamage() -> Double
}

struct Equipment: EquipmentProtocol {
    var leftHand: Item
    var rightHand: Item
    var head: Item
    var feet: Item
    var chest: Item

    func getBaseDamage() -> Double {
         leftHand.getBaseDamage() +
                rightHand.getBaseDamage() +
                head.getBaseDamage() +
                feet.getBaseDamage() +
                chest.getBaseDamage()
    }
}
